src/components/Detail/Detail.js

```diff
@@ -0,0 +1,179 @@
+//         </div>
+//       </div>
+//     </div>
+//   );
+// }
+
+// export default DetailsPage;
+//--------------------------------------------------------------------------- layout design
+import React, { useEffect, useState } from "react";
+import "./detail.css";
+
+import axios from "axios";
+
+import { useLocation } from "react-router";
+
+function DetailsPage() {
+  const location = useLocation();
+  const queryParams = new URLSearchParams(location.search);
+  const countryCode = queryParams.get("code");
+
+  const [countryData, setCountryData] = useState({});
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    axios
+      .get(`https://restcountries.com/v3.1/alpha/${countryCode}`)
+      // .get(`https://restcountries.com/v3.1/alpha/pk`)
+      .then((response) => {
+        const fetchedCountryData = response.data[0];
+        setCountryData(fetchedCountryData);
+        setLoading(false);
+      })
+      .catch((error) => {
+        console.error(error);
+        setLoading(false);
+      });
+  }, [countryCode]);
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  console.log(countryData);
+  return (
+    <div className="contents">
+      <div className="container">
+        <div className="backBtn">
+          <span className="backArrow">←</span> Back
+        </div>
+
+        <div className="wrapDetail">
+          <div className="leftFlag">
+            <img
+              src={countryData.flags?.png}
+              alt={`${countryData.name?.common} Flag`}
+              className="leftFlag"
+            />
+          </div>
+          <div className="rightDetail">
+            <h2 className="countryName">{countryData.name?.common}</h2>
+            <div className="wrapInsideDetail">
+              <div className="leftInsideDetail">
+                <p>
+                  <strong>Native Name:</strong> {countryData.name?.official}
+                </p>
+                <p>
+                  <strong>Population:</strong>{" "}
+                  {countryData.population?.toLocaleString()}
+                </p>
+                <p>
+                  <strong>Region:</strong> {countryData.region}
+                </p>
+                <p>
+                  <strong>Sub Region:</strong> {countryData.subregion}
+                </p>
+                <p>
+                  <strong>Capital:</strong>{" "}
+                  {/* array is uded to formt the captial  */}
+                  {Array.isArray(countryData.capital)
+                    ? countryData.capital.join(", ")
+                    : countryData.capital}
+                </p>
+              </div>
+              <div className="rightInsideDetail">
+                <p>
+                  <strong>Top Level Domain:</strong>{" "}
+                  {countryData.tld.join(", ")}
+                </p>
+                <p>
+                  <strong>Currencies:</strong>{" "}
+                  {Object.entries(countryData.currencies)
```

Comment:
Object.values method to directly access the currency objects' properties Object.values(countryData.currencies)

Reply:
sir in this we display the Currency in form of  i.e pakistani Rupee (Rs), and object call is invalid from this approch, like data is   USD: { name: 'US Dollar', symbol: '$' },
  EUR: { name: 'Euro', symbol: '€' },